zilliqa/src/consensus.rs

```diff
@@ -918,32 +916,34 @@ impl Consensus {
     }
 
     pub fn new_transaction(&mut self, txn: SignedTransaction) -> Result<()> {
+        let txn = txn.recover_signer()?;
```

Comment:
Suggest change to `let verified_tx` or something, as it is not obvious on first reading that this changes the type of txn here.

Reply:
To be honest, I prefer it when we shadow the previous variable once there is no longer a use for it (and indeed we can't use it after this point because `.recover_signer()` takes `self`).

@theo-zil for a tiebreaker? :stuck_out_tongue:

Reply:
I honestly only have a very weak opinion here - especially with the borrow checker ensuring `txn: SignedTransaction` won't get reused no matter what, it really doesn't matter I think. Both shadowing and making the type change explicit have their merits and their drawbacks for readability.

I think if a tiebreaker is absolutely necessary I'd side with @n-hutton here, since otherwise `recover_signer()` almost makes it look like it could just be returning the signing info rather than consuming `self` and returning the full recovered transaction.